src/components/dashjs-api-control/dashjs-api-control.tsx

```diff
@@ -96,23 +106,31 @@ export class DashjsApiControl {
   @Listen('setStream', { target: 'document' })
   setStreamEventHandler(event) {
     this.mediaUrl = event.detail;
+    setMediaURL(event.detail);
     popoverController.dismiss();
     this.loadMedia();
   }
 
   removeFunction(id: string) {
     this.selectedFunctions.set(id, undefined);
     this.selectedFunctions = new Map(this.selectedFunctions);
+    deleteLocalKey('api_functions', id);
   }
 
   updateFunction(id: string, value: any) {
     this.selectedFunctions.set(id, value);
     this.selectedFunctions = new Map(this.selectedFunctions);
     this.playerEventHandler({ type: 'function', name: id, param: value });
+    updateLocalKey('api_functions', id, value);
+  }
+
+  async resetFunctions() {
+    this.selectedFunctions = generateFunctionsMapFromList(this.functionList);
+    deleteLocalInformation('api_functions');
   }
 
   protected componentDidLoad(): void {
-    this.playerEventHandler({ type: 'autoload', autoPlay: this.element.querySelector('#autol').getAttribute('aria-checked') });
+    this.searchElement = this.element.querySelector('#searchInput');
```

Comment:
Please use a different id, it is pure luck that it isn't selecting the search input from the settings

Reply:
done